src/RedirectMiddleware.php

```diff
@@ -108,23 +110,27 @@ public function checkRedirect(
         if (!empty($options['allow_redirects']['track_redirects'])) {
             return $this->withTracking(
                 $promise,
-                (string) $nextRequest->getUri()
+                (string) $nextRequest->getUri(),
+                (string) $response->getStatusCode()
```

Comment:
I am failing to see why we need to do any casting here. As said `getUri` returns an object which MUST be casted to string, but the status code is scalar.